### DISCUSSION TITLE
Need help with making this briliant thing work

### DISCUSSION THREAD

**Discussion by freemanfl** [cedricdelpoux/gatsby-source-google-docs#254] [Q&A] [answered]
success building schema - 0.707s
12:44:14 PM: error There was an error in your GraphQL query:
12:44:14 PM: Field "page" is not defined by type "GoogleDocsFilterInput". Did you mean "path"?
12:44:14 PM: 
12:44:14 PM: 
12:44:14 PM:   GraphQLError: Field "page" is not defined by type "GoogleDocsFilterInput". Did   you mean "path"?
12:44:14 PM:   
12:44:14 PM:   - graphql-runner.ts:129 GraphQLRunner.validate
12:44:14 PM:     [repo]/[gatsby]/src/query/graphql-runner.ts:129:22

This in an error I get when trying to build my app using this plugin on vercel or netlify, I am just learning, so any help could be incredibly appreciated.

export const pageQuery = graphql`
  query Page($path: String!) {
    page: googleDocs(slug: { eq: $path }) {
      name
      cover {
        image {
          childImageSharp {
            gatsbyImageData
          }
        }
      }
      childMarkdownRemark {
        html
      }
    }
  }
`

This is a query I use from basic example

 const result = await graphql(
    `
      {
        allGoogleDocs(filter: {slug: {ne: false}}) {
          nodes {
            slug
            template
            ${fields.join(" ")}
          }
        }
      }
    `
  )
  
  This is code that gives an error in the plugin files

**Comment by cedricdelpoux** [comment]
Hello, did you try to copy templates and queries from the [basic example folder](https://github.com/cedricdelpoux/gatsby-source-google-docs/tree/master/examples/basic)?

**Reply by freemanfl** [reply]
Hi, thanks for a response! I did, and here is how the repository looks
https://github.com/freemanfl/gatsby/blob/master/src/templates/page.js

**Reply by cedricdelpoux** [reply, marked as answer]
For now `gatsby-source-google-docs` does not work with `gatsby` v5. Please try again using v4

**Reply by freemanfl** [reply]
I did try v4 and here's the error I get when trying to deploy to vercel

<html>
<body>

warning The gatsby-source-google-docs plugin has generated no Gatsby nodes. Do you need it? This could also suggest the plugin is misconfigured.
--
18:25:50.843 | success Checking for changed pages - 0.000s
18:25:50.844 | success source and transform nodes - 0.770s
18:25:50.886 | info Writing GraphQL type definitions to /vercel/path0/.cache/schema.gql
18:25:50.976 | success building schema - 0.131s
18:25:51.019 | error There was an error in your GraphQL query:
18:25:51.019 |  
18:25:51.020 | Field "page" is not defined by type "GoogleDocsFilterInput". Did you mean "path"?
18:25:51.042 |  
18:25:51.042 |  
18:25:51.042 | GraphQLError: Field "page" is not defined by type "GoogleDocsFilterInput". Did   you mean "path"?
18:25:51.042 |  
18:25:51.042 | - graphql-runner.ts:129 GraphQLRunner.validate
18:25:51.042 | [path0]/[gatsby]/src/query/graphql-runner.ts:129:22


</body>
</html>


and here's the link to the repo, I am only trying to learn adn really appreciate the help!
https://github.com/freemanfl/gatsby

**Comment by freemanfl** [comment]
It throws an error from create-pages.js file and the error says 
Field "page" is not defined by type "GoogleDocsFilterInput". Did you mean "path"?

I couldn't find the definition for "GoogleDocsFilterInput"

**Comment by freemanfl** [comment]
It works perfectly when I set the folder ID to your drive's ID, but for some reason it doesn't work wity my folder
https://drive.google.com/drive/folders/17kRSNMAVM2tP9orq3owJ-fk0BJZ_6AAm

**Reply by cedricdelpoux** [reply]
It seems like the plugo don't have access to your folder. It should not be mandatory but could you try set the privacy to public?
